main.go

```diff
@@ -338,9 +373,61 @@ func determineImageConstraints() models.ImageConstraints {
 			Version: determineNodeVersion(),
 		}
 	}
+	imageConstraints.DatabaseTypes = determineDatabaseTypes()
 	return imageConstraints
 }
 
+// determineDatabaseTypes returns a set of database types needed for tests
+func determineDatabaseTypes() map[string]struct{} {
+	databaseTypes := map[string]struct{}{}
+	if needsPostgreSQL() {
+		databaseTypes[POSTGRESQL_DB_TYPE] = struct{}{}
+	}
+	if needsMongoDB() {
+		databaseTypes[MONGO_DB_TYPE] = struct{}{}
+	}
+	return databaseTypes
+}
+
+// needsPostgreSQL returns true if tests rely on postgresql, based on these criteria:
+// -- true if a Makefile contains the text `psql`
+// -- false otherwise
+func needsPostgreSQL() bool {
+	postgresqlCheckRegexp := regexp.MustCompile(`psql`)
```

Comment:
thoughts on instead looking for postgres in the package.json/Gopkg.toml ?

Reply:
I like this suggestion a lot. Are there repos that install `pq` or `mgo` in Gopkg.toml but don't use them in tests? I'd like to avoid over-installing dbs if possible.

Reply:
Going to hold onto that idea and see what problems we run into with current version.